spec/app/domain/authentication/authn-azure/application_identity_spec.rb

```diff
@@ -0,0 +1,73 @@
+# frozen_string_literal: true
+
+require 'spec_helper'
+
+RSpec.describe Authentication::AuthnAzure::ApplicationIdentity do
```

Comment:
Missing tests:
1. Empty `role_annotations` (host without any annotations at all)
2. An annotation without `authn-azure` prefix
3. Annotations with all four azure-annotations: I think that this test should replace your current ones. This way instead of just validating the class works properly for subscription-id, you verify it for all 4.

Reply:
3. All 4 separately or individually?

Reply:
I would have them all together in 1 test

Reply:
Added 💯